### PR TITLE
Heal 76 doctors should be able to accept or reject requests

### PR DESCRIPTION
**Pull request by Shruthi23**
Addresses HEAL-76

1. Reject or accept button added if the status is pending.
2. API end point added to get all the appointments for a doctor.
3. API end point added to update the status of the appointment.

Doctors get retrieve all the appointments and view the past and upcoming appointments. The doctor can accept or reject the status of the appointment and is updated in the table.

### CODE REVIEW

File: appointments/appointment-routes.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const uuid = require("uuid");
+const AWS = require("aws-sdk");
+require("dotenv").config();
```
Comment (varunalla) [inline appointments/appointment-routes.js:1]: update readme file with details of the variables inside .env file

File: appointments/appointment-routes.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+AWS.config.update({
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  region: process.env.AWS_REGION,
+});
+const dynamodb = new AWS.DynamoDB.DocumentClient();
```
Comment (varunalla) [inline appointments/appointment-routes.js]: create a factory which returns the dynamodb client. improves re usability

File: appointments/appointment-routes.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+module.exports = (app) => {
+  app.get("/getAppointments/:doctorEmail", (req, res) => {
+    let { doctorEmail } = req.params;
```
Comment (varunalla) [inline appointments/appointment-routes.js:7]: the endpoint is not protected against JWT. before storing the email in dynamo db do you verify if the doctor is present on the blockchain?

File: client/src/components/Doctor/DoctorAppointments.tsx [hunk 4]
```diff
@@ -1,122 +1,153 @@
...
-    },
-  ];
+  const { user, role, logout } = useContext(AuthContext);
+  const [appointmentData, setAppointmentData] = useState<AppointmentDetails[]>([]);
+
+  useEffect(() => {
+    getAppointment();
+  }, []);
+  const getAppointment = async () => {
+    let resp = await axios.get('/getAppointments/' + user?.email);
```
Comment (varunalla) [inline client/src/components/Doctor/DoctorAppointments.tsx]: please use authfetch implementation which adds the jwt header directly on the front end.

**Review by varunalla — CHANGES_REQUESTED** [review]

File: appointments/appointment-routes.js (hunk 3 above)
Comment (Shruthi23) [reply appointments/appointment-routes.js:7]: What is the middleware function used to validate the JWT?
Also, the doctor would have to login to the application to enter his availability for the week. 
We can also have the admin of the HC to enter the availability fo the doctor.
Either ways, the availability table in dynamodb will have only the registered doctors

File: .DS_Store
Comment (varunalla) [inline .DS_Store:1]: remove this file

File: client/src/contracts/HealthBlock.ts
Comment (varunalla) [inline client/src/contracts/HealthBlock.ts:1]: it's fine for .ts files to exist. only delete .json files

File: client/src/contracts/factories/HCProviderHealthBlock__factory.ts
Comment (varunalla) [inline client/src/contracts/factories/HCProviderHealthBlock__factory.ts:1]: this can be deleted

**Review by varunalla — CHANGES_REQUESTED** [review]
Thank you for answering and updating the result. Please address these concerns.

File: client/src/contracts/factories/HealthBlock__factory.ts
Comment (varunalla) [inline client/src/contracts/factories/HealthBlock__factory.ts:1]: i think this file is not supposed to be deleted? or is it?

File: client/src/contracts/factories/index.ts
Comment (varunalla) [inline client/src/contracts/factories/index.ts:1]: even this file. is this file supposed to be deleted?.
to test the file has to exist or not do the following steps:
1. delete the contents of contracts folder.
2. commit
3. yarn generate type check.
4. commit the generated files.
5. now the exact number of files will surface.

File: factories/dynamodb.js
Comment (varunalla) [inline factories/dynamodb.js:1]: thank you for doing this!

**Review by varunalla — APPROVED** [review]
Please make sure the requested comments are addressed before merging. these changes if checked in will crash everyone's app

**Pull request closed without merging**